app/scripts/metamask-controller.js

```diff
@@ -2781,16 +2789,16 @@ export default class MetamaskController extends EventEmitter {
     }
 
     const serialized = await primaryKeyring.serialize();
-    const seedPhraseAsBuffer = Buffer.from(serialized.mnemonic);
 
     const accounts = await primaryKeyring.getAccounts();
     if (accounts.length < 1) {
       throw new Error('MetamaskController - No accounts found');
     }
 
     try {
-      await seedPhraseVerifier.verifyAccounts(accounts, seedPhraseAsBuffer);
-      return Array.from(seedPhraseAsBuffer.values());
```

Comment:
I'm a bit confused how/why this was encoded? `verifySeedPhrase` appears to be used in the few places we actually want to reveal the SRP as a string: [requestRevealSeedWords](https://github.com/MetaMask/metamask-extension/blob/develop/ui/store/actions.js#L228) and [seed-phrase.component.js](https://github.com/MetaMask/metamask-extension/blob/develop/ui/pages/first-time-flow/seed-phrase/seed-phrase.component.js#L33) for instance

Reply:
ok so yeah this function is only ever called by the [verifySeedPhrase action](https://github.com/MetaMask/metamask-extension/blob/develop/ui/store/actions.js#L216) in the front end which immediately converts it back to a plain text string. I can't remember if it's important that we keep it encoded when sending it from background to frontend, but currently this PR removes this flow where we convert it to a buffer and back in quick succession. I can add back the brief obfuscation if we think it's useful! @Gudahtt

Reply:
This was obfuscated here because we had, at the time, intended to leave it obfuscated for the entire process. But we stopped short of completing that work.

This does ensure that the SRP is never in string form in the background/service-worker process as well. At the time we thought this might make a difference to when we'd expect to see the SRP dropped from memory, but that was never able to be validated.

Reply:
Ok so it would be preferable to return the SRP as a `Uint8Array` here and then convert it (as was being done before) in the `verifySeedphrase` action?

Reply:
Yeah I think so. I'm not _entirely_ sure it matters, but that might be slightly safer, and would leave us in a better position to continue this obfuscation work in the future if we ever decide to.

Reply:
Fixed [here](https://github.com/MetaMask/metamask-extension/pull/17056/commits/b9b679bbedc9b66b6d2a9cef29649fb045d96630) & https://github.com/MetaMask/metamask-extension/pull/17056/commits/060a7cac0d1e866ff12b45ba6aedad092ce163d9

Reply:
It looks like it's now being sent as a Uint8Array rather than as a regular array. I would not expect this to work on Chrome because it's not valid JSON.

Reply:
This change is no longer part of this PR (as of the latest commits) because of the changes made to @metamask/eth-hd-keyring https://github.com/MetaMask/eth-hd-keyring/pull/81 which are integrated into https://github.com/MetaMask/KeyringController/pull/194 of @metamask/eth-keyring-controller and now pulled in to this PR